diff --git a/taskman/Makefile b/taskman/Makefile
index 9461d11..9461d12 100644
--- a/taskman/Makefile
+++ b/taskman/Makefile
@@ -20,6 +20,7 @@ CC        = $(CROSS)gcc
 LD        = $(CROSS)ld
 OBJCOPY   = $(CROSS)objcopy
 QSOE_RUST_TM_PROCFS ?= 0
+QSOE_RUST_TM_CRED ?= 0
 
 GCC_INCLUDE := $(shell $(CC) -print-file-name=include)
 
@@ -54,11 +55,14 @@ LIBTASKMAN_BUILD := $(abspath ../build/libtaskman)
 LIBTASKMAN_A := $(LIBTASKMAN_BUILD)/libtaskman.a
 REPO_ROOT := $(abspath ../..)
 RUST_TM_PROCFS_A := $(REPO_ROOT)/build/rust/tm-procfs/libqsoe_tm_procfs.a
+RUST_TM_CRED_A := $(REPO_ROOT)/build/rust/tm-cred/libqsoe_tm_cred.a
 
+TASKMAN_RUST_LIBS :=
+ifeq ($(QSOE_RUST_TM_CRED),1)
+TASKMAN_RUST_LIBS += $(RUST_TM_CRED_A)
+endif
 ifeq ($(QSOE_RUST_TM_PROCFS),1)
-TASKMAN_RUST_LIBS := $(RUST_TM_PROCFS_A)
-else
-TASKMAN_RUST_LIBS :=
+TASKMAN_RUST_LIBS += $(RUST_TM_PROCFS_A)
 endif
 
 .PHONY: all clean libtaskman
@@ -69,11 +73,15 @@ $(LIBTASKMAN_A): FORCE
 	    LIBC_INC=$(abspath ../../libc/include) \
 	    PICFLAG=-fno-pic \
 	    EXTRA_CFLAGS="-fno-builtin" \
+	    QSOE_RUST_TM_CRED=$(QSOE_RUST_TM_CRED) \
 	    QSOE_RUST_TM_PROCFS=$(QSOE_RUST_TM_PROCFS)
 
 $(RUST_TM_PROCFS_A): FORCE
 	@$(REPO_ROOT)/scripts/build-rust-tm-procfs-provider.sh $@
 
+$(RUST_TM_CRED_A): FORCE
+	@$(REPO_ROOT)/scripts/build-rust-tm-cred-provider.sh $@
+
 FORCE:
 
 $(BUILD)/taskman.bin: $(BUILD)/taskman.elf
